src/platform/plugins/shared/telemetry/server/plugin.ts

```diff
@@ -94,6 +94,17 @@ export interface TelemetryPluginStart {
    * @track-adoption
    */
   getIsOptedIn: () => Promise<boolean>;
```

Comment:
I'd love it if we could deprecate this API in favor of the new one.
```suggestion
   * @deprecated User {@link TelemetryPluginStart.isOptedIn$ | isOptedIn$} instead.
   */
  getIsOptedIn: () => Promise<boolean>;
```

Unless you think that it'll be needed given my other comment.